src/wallet/walletutil.cpp

```diff
@@ -14,6 +14,10 @@ fs::path GetWalletDir()
             // If the path specified doesn't exist, we return the deliberately
             // invalid empty string.
             path = "";
+        } else {
+            if (fs::is_directory(path / BaseParams().DataDir())) {
```

Comment:
This is a tricky base but after thinking a bit, I think this should be unconditional. If the user wants to use their walletdir with testnet/regtest, they should create a subdir for it in the walletdir.
Otherwise this will still result in wallet directory collision by default if `-walletdir` is specified in bitcoin.conf.
We really want to avoid the case where accidentally, say, a regtest wallet is used on mainnet or vice versa.

Reply:
After discussion on IRC, I realized maybe we don't want to do this at all and just wait for #10996 to go in. This will remove the issue by making it possible to set a separate walletdir per network.

Reply:
Agreed, #10996 or something similar would be much cleaner. Removed.